src/app/components/Menu/Menu.tsx

```diff
@@ -1,14 +1,47 @@
 'use client';
-import React, { useEffect, useState } from 'react';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
 import Image from 'next/image';
 import './Menu.scss';
 import Button from '../Button/Button';
 import { useRouter } from 'next/navigation';
+import throttle from 'lodash.throttle';
```

Comment:
Co to daje?

Reply:
throttle powoduje, że nie odpala funkcji za każdą zmianą osi y